File: Inheritance/1.py
class Class1(object):
    def method1(self): return("method1")

class Class2(Class1):
    def method2(self): return("method2")

c1 = Class1()
c2 = Class2()

# You can print the class name of the instance the output came from with it.
print(c1, c1.method1())
print(c2, c2.method1())
print(c2, c2.method2())

File: Inheritance/1.rb
class Class1
    def method1 
        "method1"
    end
end

class Class2 < Class1
    def method2 
        "method2"
    end
end

c1 = Class1.new
c2 = Class2.new

# You can print the class name of the instance the output came from with it.
p c1, c1.method1
p c2, c2.method1
p c2, c2.method2